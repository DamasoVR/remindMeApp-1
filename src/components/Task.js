/**
 * RemindMe App
 * Task Component
 */

import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  TouchableHighlight,
  Image,
  Dimensions,
  View
} from 'react-native';
import Colors from './../theme/colors';

const screen = Dimensions.get('window');

export default class HomeScreen extends Component<{}> {

  constructor(props){
    super(props);
  }

  render() {

    const { id, title } = this.props;
    let { completed } = this.props;
    let icon = completed ? require('./../images/icon-checked.png') : require('./../images/icon-circle.png');

    return (
        <TouchableHighlight
            style={styles.touchableArea}
            onPress={ () => { this.props.toggleTask(id) } }>
          <View style={styles.container}>
            <Image
              style={styles.iconUnchecked}
              source={icon}/>
            <Text style={ completed ? styles.taskCompletedText : styles.taskText }>{title}</Text>
          </View>
        </TouchableHighlight>
    );
  }
}

// TODO: Create Theme styles (Buttons, Global Components, etc) so it can be imported on each component as required.
const styles = StyleSheet.create({
touchableArea:{
  height: 60
},
  container: {
    flex: 1,
    flexDirection: 'row',
    height: 60,
    padding: 15,
    borderBottomWidth: 2,
    borderBottomColor: Colors.separator
  },
  iconUnchecked: {
    width: 20,
    height: 20,
    marginRight: 10,
    marginTop: 4
  },
  taskText: {
    fontSize: 21,
    fontFamily: 'Avenir'
  },
  taskCompletedText: {
    fontSize: 21,
    fontFamily: 'Avenir',
    color: Colors.gray
  }
});
