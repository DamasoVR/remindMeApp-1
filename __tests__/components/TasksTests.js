import 'react-native';
import React from 'react';
import Task from '../../src/components/Task';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

let component;
const toogleTaskMock = jest.fn();

describe('Task Component - ', () => {

  it('should render correctly', () => {
    component = renderer.create(
      <Task
        id={1}
        title={'Test Components'}
        toggleTask={toogleTaskMock}
        completed={false}/>
    ).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('should render completed state correctly', () => {
    component = renderer.create(
      <Task
        id={1}
        title={'Test Components'}
        toggleTask={toogleTaskMock}
        completed={true}/>
    ).toJSON();
    expect(component).toMatchSnapshot();
  });

});
