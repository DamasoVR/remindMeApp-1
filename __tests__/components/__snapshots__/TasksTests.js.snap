// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task Component -  should render completed state correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 60,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "borderBottomColor": "rgb(231,231,231)",
        "borderBottomWidth": 2,
        "flex": 1,
        "flexDirection": "row",
        "height": 60,
        "padding": 15,
      }
    }
  >
    <Image
      source={
        Object {
          "process": [Function],
        }
      }
      style={
        Object {
          "height": 20,
          "marginRight": 10,
          "marginTop": 4,
          "width": 20,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "rgb(164, 164, 164)",
          "fontFamily": "Avenir",
          "fontSize": 21,
        }
      }
    >
      Test Components
    </Text>
  </View>
</View>
`;

exports[`Task Component -  should render correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 60,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "borderBottomColor": "rgb(231,231,231)",
        "borderBottomWidth": 2,
        "flex": 1,
        "flexDirection": "row",
        "height": 60,
        "padding": 15,
      }
    }
  >
    <Image
      source={
        Object {
          "process": [Function],
        }
      }
      style={
        Object {
          "height": 20,
          "marginRight": 10,
          "marginTop": 4,
          "width": 20,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "Avenir",
          "fontSize": 21,
        }
      }
    >
      Test Components
    </Text>
  </View>
</View>
`;
